fbinternal/teravm/fabfile.py

```diff
@@ -0,0 +1,654 @@
+    except Exception:
+        fastprint("Image %s not found. Not updating AGW \n" % hash)
+        return
+    _fetch_image("ag", "gateway/%s" % image)
+
+    with cd("/tmp/images"):
+        run("tar -xzf %s" % image)
+        # --fix-broken to avoid the case where a previous manual
+        # install didn't leave missing libraries.
+        sudo(
+            "apt --fix-broken -y install -o "
+            'Dpkg::Options::="--force-confnew" --assume-yes --force-yes'
+        )
+        sudo("apt-get update -y")
+        sudo("apt-get autoremove -y")
+        sudo(
+            "apt --fix-broken -y install -o "
+            'Dpkg::Options::="--force-confnew" --assume-yes --force-yes'
+        )
+        sudo("dpkg --force-confnew -i magma*.deb")
+        sudo("apt-get install -f -y")
+        sudo("systemctl stop magma@*")
+        sudo("systemctl restart magma@magmad")
+
+
+def upgrade_teravm_feg(setup, hash, key_filename=DEFAULT_KEY_FILENAME):
+    """
+    Upgrade teravm feg to the image with the given hash.
+
+    hash: a hash to identify what images to pull from s3 bucket and use
+    for upgrading. If None, try find the most recent hash.
+
+    key_filename: path to where the private key is for authorized-key based
+    ssh. The public key counterpart needs to in the authorized_keys file on
+    the remote host. IIf empty file name is passed, password-based ssh will
+    work instead. This can be used if the script is run manually.
+    """
+    err = None
+    fastprint("Upgrade teraVM FEG to %s" % hash)
+    _setup_env("magma", VM_IP_MAP[setup]["feg"], key_filename)
+
+    with cd("/var/opt/magma/docker"), settings(abort_exception=FabricException):
+        sudo("docker-compose down")
+        sudo("cp docker-compose.yml docker-compose.yml.backup")
+        sudo("cp .env .env.backup")
+        sudo('sed -i "s/IMAGE_VERSION=.*/IMAGE_VERSION=%s/g" .env' % hash)
+        if len(_check_disk_space()) != 0:
+            fastprint("Disk space alert: cleaning docker images\n")
+            sudo("docker system prune --all  --force")
+        try:
+            # TODO: obtain .yml file from jfrog artifact instead of git master
+            sudo("wget -O docker-compose.yml %s" % FEG_DOCKER_COMPOSE_GIT)
+            sudo("docker-compose up -d")
+        except Exception:
+            err = (
+                "Error during install of version {}. Maybe the image "
+                "doesn't exist. Reverting to the original "
+                "config\n".format(hash)
+            )
+            fastprint(err)
+            with hide("running", "stdout"):
+                sudo("mv docker-compose.yml.backup docker-compose.yml")
+                sudo("mv .env.backup .env")
+                sudo("docker-compose up -d")
+    return err
+
+
+def upgrade_teravm_cloud_vm(setup, name, hash, key_filename=DEFAULT_KEY_FILENAME):
+    # TODO: implement in case local orc8r is needed
+    fastprint("Upgrade teraVM Cloud: Not implemented for current version")
```

Comment:
This comment reminded me that we use staging right now. Do we want to keep it that way? migrate to its own namespace?

Reply:
right now it will be staging. Once we stabilize the tests and get this rolling the idea is ITL will set up a new orc8r and we will update the fab script to include that update too